river/metrics/cluster/silhouette.py

```diff
@@ -0,0 +1,113 @@
+import math
+
+from river import utils
+
+from . import base
+
+__all__ = ["Silhouette"]
+
+
+class Silhouette(base.InternalClusMetric):
```

Comment:
Is this metric the one whose definition differs from the traditional batch version? If so, I advise adding a brief note warning the user.

Reply:
It is indeed different. I have also added an explanation on why and how is it different!